### PR TITLE
[waiting for golang v1.23 release][cleanup] use golang 1.22: and bump kube to 0.30.3 and etcd to v3.5.14

### PR DESCRIPTION
**Pull request by pacoxu**
including https://github.com/coredns/coredns/pull/6794

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -276,12 +276,12 @@ github.com/yuin/goldmark v1.1.27/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9de
...
-go.etcd.io/etcd/client/pkg/v3 v3.5.13 h1:RVZSAnWWWiI5IrYAXjQorajncORbS0zI48LQlE2kQWg=
-go.etcd.io/etcd/client/pkg/v3 v3.5.13/go.mod h1:XxHT4u1qU12E2+po+UVPrEeL94Um6zL58ppuJWXSAB8=
-go.etcd.io/etcd/client/v3 v3.5.13 h1:o0fHTNJLeO0MyVbc7I3fsCf6nrOqn5d+diSarKnB2js=
-go.etcd.io/etcd/client/v3 v3.5.13/go.mod h1:cqiAeY8b5DEEcpxvgWKsbLIWNM/8Wy2xJSDMtioMcoI=
+go.etcd.io/etcd/api/v3 v3.5.14 h1:vHObSCxyB9zlF60w7qzAdTcGaglbJOpSj1Xj9+WGxq0=
+go.etcd.io/etcd/api/v3 v3.5.14/go.mod h1:BmtWcRlQvwa1h3G2jvKYwIQy4PkHlDej5t7uLMUdJUU=
+go.etcd.io/etcd/client/pkg/v3 v3.5.15 h1:fo0HpWz/KlHGMCC+YejpiCmyWDEuIpnTDzpJLB5fWlA=
+go.etcd.io/etcd/client/pkg/v3 v3.5.15/go.mod h1:mXDI4NAOwEiszrHCb0aqfAYNCrZP4e9hRca3d1YK8EU=
+go.etcd.io/etcd/client/v3 v3.5.14 h1:CWfRs4FDaDoSz81giL7zPpZH2Z35tbOrAJkkjMqOupg=
+go.etcd.io/etcd/client/v3 v3.5.14/go.mod h1:k3XfdV/VIHy/97rqWjoUzrj9tk7GgJGH9J8L4dNXmAk=
```
Comment (liangyuanpeng) [inline go.sum]: Is this expected?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/coredns/coredns
 
-go 1.21
+go 1.22.5
```
Comment (SuperQ) [inline go.mod:3]: We would like to support the latest two Go releases.

**Review by SuperQ — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (pacoxu) [reply go.mod:3]: > go: k8s.io/api@v0.30.3 requires go >= 1.22.0 (running go 1.21.0; GOTOOLCHAIN=go1.21.0)

This is required by v0.30.3. 

Or we may wait for golang 1.23 release and then bump.

File: go.sum (hunk 1 above)
Comment (pacoxu) [reply go.sum]: following k8s: updated to all etcd related to 3.5.14.

**Comment by pacoxu** [comment]
[CI failure](https://github.com/coredns/coredns/actions/runs/10160155036/job/28095901921?pr=6799) is a different issue: a quick fix in https://github.com/coredns/coredns/pull/6802

File: go.mod (hunk 2 above)
Comment (SuperQ) [reply go.mod:3]: Yes, we will need to hold back on k8s.io dependencies until 1.23.

**Comment by SuperQ** [comment]
1.23 is out, and I have updated the build.

This should be good to rebase.

**Comment by SuperQ** [comment]
Superseded by https://github.com/coredns/coredns/pull/6840

**Pull request closed without merging**